libopenage/unit/producer.cpp

```diff
@@ -534,12 +535,53 @@ void BuildingProducer::initialise(Unit *unit, Player &player) {
 		unit->give_ability(std::make_shared<AttackAbility>());
 	}
 
+	// dropsite attribute
+	std::vector<game_resource> accepted_resources = get_accepted_resources();
+	if (accepted_resources.size() != 0) {
+		unit->add_attribute(std::make_shared<Attribute<attr_type::dropsite>>(accepted_resources));
+	}
+
 	// building can train new units and ungarrison
 	unit->give_ability(std::make_shared<SetPointAbility>());
 	unit->give_ability(std::make_shared<TrainAbility>());
 	unit->give_ability(std::make_shared<UngarrisonAbility>());
 }
 
+std::vector<game_resource> BuildingProducer::get_accepted_resources() {
+	//TODO use a more general approach instead of hard coded ids
+
+	auto id_in = [id = this->id()](std::initializer_list<int> ids){
+		return std::any_of(ids.begin(), ids.end(), [=](int n){ return n == id; });
```

Comment:
you can use `this->id()` in the body and capture by value `=` in the lambda, then the `id = this->id()` is no longer necessary.